lib/config/providers/chat_provider.dart

```diff
@@ -745,6 +752,118 @@ class ChatNotifier extends Notifier<ChatState> {
 
     await ref.read(groupsProvider.notifier).updateGroupActivityTime(groupId, now);
   }
+
+  /// Get cached DMChatData for a group, or load it if not cached
+  Future<DMChatData?> getDMChatData(String groupId) async {
+    // Return cached data if available
+    if (state.isDMChatDataCached(groupId)) {
+      return state.getDMChatData(groupId);
+    }
+
+    if (_dmChatDataLoadingFutures.containsKey(groupId)) {
+      return _dmChatDataLoadingFutures[groupId];
+    }
+
+    // Load data asynchronously and cache it
+    try {
+      _logger.info('Loading DMChatData for group: $groupId');
+      final future = _loadDMChatData(groupId);
+      _dmChatDataLoadingFutures[groupId] = future;
+      final dmChatData = await future;
+      _dmChatDataLoadingFutures.remove(groupId);
+
+      // Update state with cached data
+      state = state.copyWith(
+        dmChatDataCache: {
+          ...state.dmChatDataCache,
+          groupId: dmChatData,
+        },
+      );
+
+      return dmChatData;
+    } catch (e, st) {
+      _logger.severe('Failed to load DMChatData for group $groupId', e, st);
+      _dmChatDataLoadingFutures.remove(groupId);
+      // Cache null result to avoid repeated failed attempts
+      state = state.copyWith(
+        dmChatDataCache: {
+          ...state.dmChatDataCache,
+          groupId: null,
+        },
+      );
+      return null;
+    }
+  }
+
+  /// Preload DMChatData for a group to ensure it's available immediately
+  Future<void> preloadDMChatData(String groupId) async {
+    if (!state.isDMChatDataCached(groupId)) {
+      await getDMChatData(groupId);
+    }
+  }
+
+  /// Get cached DMChatData synchronously (returns null if not cached)
+  DMChatData? getCachedDMChatData(String groupId) {
+    return state.getDMChatData(groupId);
+  }
+
+  /// Clear cached DMChatData for a specific group
+  void clearDMChatDataForGroup(String groupId) {
+    if (state.isDMChatDataCached(groupId)) {
+      final newCache = Map<String, DMChatData?>.from(state.dmChatDataCache);
+      newCache.remove(groupId);
+      state = state.copyWith(dmChatDataCache: newCache);
+    }
+  }
+
+  /// Helper method to load DMChatData using the NotifierProviderRef with retry strategy
+  Future<DMChatData?> _loadDMChatData(String groupId) async {
+    const maxRetries = 2;
+    int attempt = 0;
+
+    while (attempt <= maxRetries) {
+      try {
+        final otherMember = ref.read(groupsProvider.notifier).getOtherGroupMember(groupId);
+
+        if (otherMember != null) {
+          final user = await wn_users_api.getUser(pubkey: otherMember.publicKey);
```

Comment:
Why not using user profile provider? it looks like the `getUserProfile ` does the same but it is tested (alles the get user method, then transforms the response to a UserProfile object